tests/pytests/test_expire.py

```diff
@@ -499,6 +499,36 @@ def testDocWithLongExpiration(env):
     conn.execute_command('EXPIRE', 'doc:1', '30000')
     env.expect('FT.SEARCH', 'idx', 'hello', 'NOCONTENT').apply(sort_document_names).equal([2, 'doc:1', 'doc:2'])
 
+def testSeekToExpirationChecks(env):
+    # We want to cover the IndexReader_ReadWithSeeker function
+    conn = getConnectionByEnv(env)
+    conn.execute_command('DEBUG', 'SET-ACTIVE-EXPIRE', '0')
+    conn.execute_command('FT.CREATE', 'idx', 'SCHEMA', 'x', 'TEXT', 'y', 'TEXT')
+    conn.execute_command('HSET', 'doc:0', 'x', 'hello', 'y', 'foo') # doc:expire internal id is 1001
+    # inverted index state
+    # 'hello': [1]
+    # 'foo': [1]
+    for i in range(1, 1001):
+        conn.execute_command('HSET', f'doc:{i}', 'x', 'hello', 'y', 'world')
+        # important we expire now since that assigns a new doc id for doc:expire - 1002
+        conn.execute_command('HPEXPIRE', f'doc:{i}', '1', 'FIELDS', '1', 'y') # doc:expire internal id is 5
+    conn.execute_command('HSET', 'doc:1001', 'x', 'hello', 'y', 'world')
+    # inverted index state
+    # 'hello': ['doc:0', 'doc:1', , ..., 'doc:1000', 'doc:1001']
+    # 'world': ['doc:1', , ..., 'doc:1000', 'doc:1001']
+    # 'foo': ['doc:0']
+
+    # expected flow
+    # - hello reader starts with doc:0
+    # - world reader starts with doc:1
+    # - intersect iterator reads doc:0 and tries to skip to it in world reader
+    # - world reader should skip to doc:1001 since all the other docs will be expired
```

Comment:
isn't `hello` reader should skip to `1001` since `world` reader has been called with IR_SkipTo with `1`, uses `IndexReader_ReadWithSeeker` and returns `1001`?